src/ripple/app/ledger/LedgerHistory.cpp

```diff
@@ -312,9 +312,12 @@ leaves (SHAMap const& sm)
     return v;
 }
 
-void LedgerHistory::handleMismatch (
+void
+LedgerHistory::handleMismatch(
     LedgerHash const& built,
     LedgerHash const& valid,
+    boost::optional<uint256> const& builtConsensusHash,
```

Comment:
At first, I thought this could never be unseated, that it'd be better suited as a `uint256 const&`. I realize that it can be called with an actual `optional` so leaving it like this is fine, but I am still not sure under what circumstances it can be unseated?

Reply:
Its fairly indirect, but my thinking was that `sfConsensusHash` is technically an optional field in `STValidation`, so even though `rippled` will now populate it, the protocol doesn't strictly require it. So then I ended up keeping `validatedConsensusHash` as an optional all the way through. 

The `builtConsensusHash` will never be unseated after this change, but I left it as optional to match `validatedConsensusHash`.